### ISSUE TITLE
[0.1001.0] Multiplayer Campaign - Picked up mission/money spending deyncs

### ISSUE THREAD

**Issue by Kaillera** [Regalis11/Barotrauma#3020]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Not sure if the issue is caused by #2942, but when players reconnects to their bodies while in an outpost, money spent and picked up missions are not syncing to every players. I was spending money and picking up missions but the clients who didn't had to reconnect was not able to see any picked up missions when undocking nor and saw the Marks were unused.

When clients that did reconnected and picked up missions, selecting a mission when other clients (and server apparently) didn't register them as well causes Missing Entity of Mission Item or Target.

**Version**
0.1001.0

**Comment by ekqvist** [comment]
I've tried to reproduce the money desync issue on my end, but haven't succeeded so far. I tried to

1. Join the game with Client 1 (C1) and Client 2 (C2)
2. Disconnect with C2
3. Buy supplies, sell supplies, purchase and switch the sub, and buy sub upgrades with C1
4. Reconnect with C2
5. Use console command `campaigninfo` with C1 and C2 to see that the money value is the same for both clients

I did not, however, have repeating prompts to download the save file when testing this so I'm not sure if it's related to reproducing this issue. I also picked up a mission while C2 was disconnected and C2 did get kicked for Missing Entity later.
